dali/operators/reader/fits_reader_op.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_FITS_READER_OP_H_
+#define DALI_OPERATORS_READER_FITS_READER_OP_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/operators/reader/loader/fits_loader.h"
+#include "dali/operators/reader/reader_op.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/util/crop_window.h"
+
+namespace dali {
+
+template <typename Backend, typename Target>
+class FitsReader : public DataReader<Backend, Target> {
+ public:
+  explicit FitsReader(const OpSpec& spec) : DataReader<Backend, Target>(spec) {}
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  USE_READER_OPERATOR_MEMBERS(Backend, Target);
+  using DataReader<Backend, Target>::GetCurrBatchSize;
+  using DataReader<Backend, Target>::GetSample;
+  using Operator<Backend>::spec_;
+
+
+  bool SetupImpl(std::vector<OutputDesc>& output_desc, const Workspace& ws) override {
+    // If necessary start prefetching thread and wait for a consumable batch
+    DataReader<Backend, Target>::SetupImpl(output_desc, ws);
+
+    int batch_size = GetCurrBatchSize();
+    const auto& file_0 = GetSample(0);
+    DALIDataType output_type = file_0.get_type();
+    int ndim = file_0.get_shape().sample_dim();
+    TensorListShape<> sh(batch_size, ndim);
+
+    // TODO: implement checking that all images have same dimensions
+    // also all general calculations for all images such as roi
```

Comment:
In the initial case it might be ok to assume uniform batch but maybe there shouldn't be such constraint in general and we should support image of different sizes.

Also, you need to peek the shape of the sample and set it for all element of `sh` so the executor can correctly resize the buffers. Now it will probably have some random contents (we just resized the internal vector to `batch_size * ndim` elements), so this is probably a bug.